ipam.py

```diff
@@ -98,9 +104,18 @@ def execute(self):
         :return: CNI ipam dictionary for ADD, None for DEL.
         """
         if self.command == "ADD":
-            # Assign an IP address for this workload.
-            _log.info("Assigning address to workload: %s", self.workload_id)
-            ipv4, ipv6 = self._assign_address(handle_id=self.workload_id)
+            if self.ip:
+                # The user has specifically requested an IP (v4) address.
+                _log.info("User assigned address: %s for workload: %s",
+                          self.ip,
+                          self.workload_id)
+
+                ipv4 = self._assign_existing_address()
+                ipv6 = None
+            else:
+                # Assign an IP address for this workload.
```

Comment:
"Auto-assign an IP address for this workload"

Should clean up the terminology in this module now to be clear when we're auto-assigning and when we're not.